train.py

```diff
@@ -216,7 +216,7 @@ def loss_fn(pred, labels):
         whole_model = model_cls.from_model_args(model_config)
 
     # apply fp8 linear module swap
-    if job_config.training.enable_fp8_linear:
+    if SM90OrLater and job_config.training.enable_fp8_linear:
```

Comment:
hmmm I think we don't need to expose this to train.py?

We can try to check this directly in `build_fp8_linear` function, and if its not SM90OrLater, we can throw a warning and don't swap the linear layers? similar to other fp8 APIs in float8_linear.py, only apply when SM90OrLater, otherwise throw warnings

Reply:
got you. will move it to fp8 api with warnings

Reply:
Alternatively you could consider calling it `maybe_build_fp8_linear` and read `enable_fp8_linear` from `job_config`. Within that function, you can do logging to let user know if build is successful.

Reply:
created `maybe_build_fp8_linear` and `maybe_precompute_fp8_dynamic_scale_for_fsdp`. config and SM90 check are moved from train.py into maybe_ functions